src/types.ts

```diff
@@ -32,14 +32,73 @@ export type DocStyle = 'jsdoc' | 'tomdoc'
 
 export type Arrayable<T> = T | readonly T[]
 
+export type ResultNotFound = {
+  found: false
+  path?: undefined
+}
+
+export type ResultFound = {
+  found: true
+  path: string | null
+}
+
+export type ResolvedResult = ResultNotFound | ResultFound
+
+export type ResolverResolve = (
+  modulePath: string,
+  sourceFile: string,
+  config: unknown,
+) => ResolvedResult
+
+export type ResolverResolveImport = (
+  modulePath: string,
+  sourceFile: string,
+  config: unknown,
+) => string | undefined
```

Comment:
Is it possible to make the `config` generic? In case some resolver library authors want to use these types.